modules/contact/test/tests/fieldsplit/tests

```diff
@@ -11,11 +11,10 @@
   []
   [mortar-FS]
     type = 'Exodiff'
-    input = 'frictional_mortar_FS.i'
-    exodiff = 'frictional_mortar_FS_out.e'
+    input = 'frictionless_mortar_FS.i'
+    exodiff = 'frictionless_mortar_FS_out.e'
     abs_zero = 1e-8
-    mesh_mode = REPLICATED
-    skip = 'Weighted mortar'
-    requirement = 'The system shall allow split preconditioning of frictional mortar contact.'
+    requirement = 'The system shall allow split preconditioning of frictionless mortar contact.'
+    petsc_version = '>= 3.16.6' # early versions of petsc can't handle change in sparsity pattern in field split
```

Comment:
in my limited understand of contact, frictionless and frictional would be different.
can you document why you are making that change and is it necessary for you to? does frictional not make sense here?

Reply:
or does it just not matter

Reply:
I guess I did it because the node-face version was frictionless

Reply:
I can work on changing it back if you'd prefer

Reply:
Or I think I would add a frictional test and keep what is now the frictionless test

Reply:
I didn't think much about changing the requirement given that the test was being skipped coming into this

Reply:
Well depends on if it matters wrt to the subject at hand

Reply:
With mortar contact you don't actually need to use the `ContactSplit` object nor the contact data members on `DM_Moose`. You can just rely on splitting by variables, whereas for node-face you don't have Lagrange multipliers so you need to split based on displacement interior and contact interface dofs, and that *does* use the `ContactSplit` object and the contact data members on `DM_Moose`. So in terms of using the contact lines of code I've touched in this PR, only the node-face test does that.

Reply:
It doesn't cost me much to add a frictional test. I'll just do that now

Reply:
Didn't look at the code... Just wanted to add that one may want to use field split with penalty mortar contact, without LMs.

Reply:
For which we still have the ContactSplit which we test in the node face test. 

We don’t have penalty mortar mechanical contact do we?

Reply:
We should have penalty mortar contact soon. 

I assume using the condensed LMs approach would make the same usage of field split as the node face test you pointed out too.